source/extensions/bootstrap/internal_listener/client_connection_factory.h

```diff
@@ -35,6 +37,11 @@ class InternalClientConnectionFactory : public Network::ClientConnectionFactory,
   // TODO(lambdai): make it friend to only bootstrap extension.
   static ThreadLocal::TypedSlot<Bootstrap::InternalListener::ThreadLocalRegistryImpl>*
       registry_tls_slot_;
+  // The buffer size of the internal client connection. It has to be static because of:
+  // Client factory establishes the buffer while internal listener registry accepts the
+  // client connection. Thus the ownership is not followed - client factory needs to
+  // access buffer_size config of listener registry.
+  static uint32_t buffer_size_;
```

Comment:
This seems undesired.
Does this imply that there is a single `buffer_size_` for all InternalListeners?

Reply:
Yeah, the design is to have this config for all internal listeners.  There is no need to tweak this for each one.

Reply:
If there is only a slight difference in making it per-listener I think it might be better to do that. Surely this fits reasonably in the Listener config?

Reply:
Thanks for the comments. Per Kuat's comments below: "custom sizing for different sockets, that can be delayed until we actually have an ability to use multiple internal listener registries".